app/models/document.rb

```diff
@@ -55,8 +60,12 @@ def document_type_label
   end
 
   def set_display_name
-    return if display_name
+    return if display_name.present?
 
     self.display_name = upload.attachment.filename
   end
+
+  def tax_return_belongs_to_client
+    errors.add(:tax_return) unless tax_return.blank? || tax_return.client == client
```

Comment:
Is it possible to add a specific error message so that it's clear what the problem is?

Reply:
Will do